sentinel/lib/messages.js

```diff
@@ -1,228 +1,115 @@
-        addMessage({
-            doc: doc,
-            message: message,
-            phone: self.getRecipientPhone(doc, 'parent_phone'),
-            templateContext: options
-        });
+        // otherwise, use the configured messages (deprecated)
+        var messages = configuration.messages || configuration.message;
+        if (!_.isArray(messages)) {
+            console.warn('Message property should be an array. Please check your configuration.');
+            return '';
+        }
+        if (!messages.length) {
+            console.warn('Message property array was empty. Please check your configuration.');
+            return '';
+        }
+        // default to first item in messages array in case locale match fails
+        var message = _.findWhere(messages, { locale: locale || 'en' }) || messages[0];
+        return (message.content && message.content.trim()) || '';
     },
-    addReply: function(doc, message, options) {
-        var self = module.exports;
-        addMessage({
-            doc: doc,
-            message: message,
-            phone: self.getRecipientPhone(doc, 'clinic'),
-            templateContext: options
-        });
+    /*
+     * Return false when the recipient phone matches the denied list.
+     *
+     * outgoing_deny_list is a comma separated list of strings. If a string in
+     * that list matches the beginning of the phone then we set up a response
+     * with a denied state. The pending message process will ignore these
+     * messages and those reports will be left without an auto-reply. The
+     * denied messages still show up in the messages export.
+     *
+     * @param {String} from - Recipient phone number
+     * @returns {Boolean}
+     */
+    isOutgoingAllowed: from => {
+      const conf = config.get('outgoing_deny_list') || '';
+      if (!from) {
+        return true;
+      }
+      if (module.exports.isMessageFromGateway(from)) {
+        return false;
+      }
+      return _.every(conf.split(','), s => {
+        // ignore falsey inputs
+        if (!s) {
+          return true;
+        }
+        // return false if we get a case insensitive starts with match
+        return from.toLowerCase().indexOf(s.trim().toLowerCase()) !== 0;
+      });
     },
-    addErrors: function(doc, errors) {
-        var self = module.exports;
-        _.each(errors, function(error) {
-            self.addError(doc, error);
-        });
+    addErrors: function(config, doc, errors) {
+        const self = module.exports;
+        
+        errors.forEach(error => self.addError(doc, error));
+        
+        let reply;
+        if (config.validations.join_responses) {
+            const msgs = [];
+            errors.forEach(err => {
+                if (err.message) {
+                    msgs.push(err.message);
+                } else if (err) {
+                    msgs.push(err);
+                }
+            });
+            reply = msgs.join('  ');
```

Comment:
Is this 2 spaces deliberately?

Reply:
I copied it from three different places in the code, all had two spaces, so I guess it's deliberate...